pkg/operator/staticpod/controller/installer/installer_controller.go

```diff
@@ -478,19 +479,26 @@ func (c *InstallerController) ensureInstallerPod(nodeName string, operatorSpec *
 	pod.Spec.Containers[0].Image = c.installerPodImageFn()
 	pod.Spec.Containers[0].Command = c.command
 
+	if c.configMaps[0].Optional {
+		return fmt.Errorf("pod configmap %s is required, cannot be optional", c.configMaps[0].Name)
+	}
 	args := []string{
 		"-v=4", // TODO: Make this configurable?
 		fmt.Sprintf("--revision=%d", revision),
 		fmt.Sprintf("--namespace=%s", pod.Namespace),
-		fmt.Sprintf("--pod=%s", c.configMaps[0]),
+		fmt.Sprintf("--pod=%s", c.configMaps[0].Name),
 		fmt.Sprintf("--resource-dir=%s", hostResourceDirDir),
 		fmt.Sprintf("--pod-manifest-dir=%s", hostPodManifestDir),
 	}
-	for _, name := range c.configMaps {
-		args = append(args, fmt.Sprintf("--configmaps=%s", name))
+	for _, cm := range c.configMaps {
+		if !cm.Optional {
+			args = append(args, fmt.Sprintf("--configmaps=%s", cm.Name))
```

Comment:
These are args to https://github.com/openshift/library-go/blob/master/pkg/operator/staticpod/installerpod/cmd.go#L131 which does not tolerate the idea of optional resources.  You need optional resources, so that information must be plumbed through and those which are *optional* should be tolerated as missing.

Reply:
Ah, I see now. Will fix.